server.go

```diff
@@ -1102,7 +1067,7 @@ func (sp *serverPeer) OnBlock(p *peer.Peer, msg *wire.MsgBlock, buf []byte) {
 func (sp *serverPeer) OnInv(p *peer.Peer, msg *wire.MsgInv) {
 	// Ban peers sending empty inventory requests.
 	if len(msg.InvList) == 0 {
-		sp.server.BanPeer(sp)
+		sp.banManager.BanPeer(p)
```

Comment:
`sp.s.banManager.BanPeer(p)` if suggested change is made.  Or you could even introduce a helper like:

```Go
func (sp *serverPeer) banPeer() {
	sp.s.banManager.BanPeer(sp.Peer)
}
```

Then all these ban calls would be just turn into `sp.banPeer()`